### ISSUE TITLE
Song caché: Doesn't check for removed songs.

### ISSUE THREAD

**Issue by Lord-Kamina** [performous/performous#386]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Change my song folder to one I use for testing.

### What did you expect to see?

Songs from my new song folder.

### What did you see instead?

Songs from the new folder, plus songs from the old one.
I expected a segfault or error, actually, but this at least didn't happen in my case because the metadata json stores absolute paths. Nonetheless, this is definitely an issue.

### Output of `performous --version`: (_What version of Performous are you using?_)

```
(paste your output here)
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
(paste your output here)
```

**Comment by nieknooijens** [comment]
workaround: delete the cache file in your home folder.

**Comment by Baklap4** [comment]
I need some more info regarding this as the cache does check for removed songs:

```
for(auto const& song : jsonRoot.as_array()) {
    	struct stat buffer;
    	if(stat(song.at("TxtFile").as_string().c_str(), &buffer) == 0) {
    		std::shared_ptr<Song> realSong(new Song(song));
    		m_songs.push_back(realSong);
    	}  	
}
```
@Lord-Kamina did you mean by 'removed' songs changing the song directory (within your config.xml) or by adding it as a parameter? While the other folder is still available on that hard drive? Because in my opinion it's not a deleted song as it's still available from a source.

I think the desired behavior would be to check if the txt file exists (like it happens now) AND to check if that file is in one of the configured song directories.

**Comment by Lord-Kamina** [comment]
I changed it in the config.xml, and I see your point 

I agree re:the desired behavior.

**Comment by Lord-Kamina** [comment]
@nieknooijens that's what I did for subsequent tests.

**Comment by Baklap4** [comment]
Aight will make a pr tonight for this behavior 👍

**Comment by Baklap4** [comment]
Fix comitted as per #405
